dartsim/src/SimulationFeatures.cc

```diff
@@ -15,26 +15,33 @@
  *
 */
 
+#include <unordered_map>
+
 #include <dart/collision/CollisionObject.hpp>
 #include <dart/collision/CollisionResult.hpp>
 
-#include "SimulationFeatures.hh"
+#include <ignition/common/Profiler.hh>
+
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/eigen3/Conversions.hh>
 
-#include "ignition/common/Profiler.hh"
 #include "ignition/physics/GetContacts.hh"
 
+#include "SimulationFeatures.hh"
+
 namespace ignition {
 namespace physics {
 namespace dartsim {
 
 void SimulationFeatures::WorldForwardStep(
     const Identity &_worldID,
-    ForwardStep::Output & /*_h*/,
+    ForwardStep::Output & _h,
     ForwardStep::State & /*_x*/,
     const ForwardStep::Input & _u)
 {
   IGN_PROFILE("SimulationFeatures::WorldForwardStep");
   auto *world = this->ReferenceInterface<DartWorld>(_worldID);
+  // TODO(adlarkin) sanity check to make sure world is not nullptr?
```

Comment:
If `world` is `nullptr`, should we just `return` from the function early?

Reply:
Added this check in https://github.com/ignitionrobotics/ign-physics/pull/223/commits/2df9ea0c290f75ea72a9781ba003ca0b84d15ad1